### DISCUSSION TITLE
Write arduino every 1 second while condition is met

### DISCUSSION THREAD

**Discussion by vdtang3** [bonsai-rx/bonsai#2058] [Q&A]
Hi,

I'm trying to send a command to an arduino via SerialStringWrite every 1 second while a condition is being met (in this case when the animal is in a certain part of a VR corridor). I've tried to do this by creating a conditional on a SubscribeSubject (z pos), and adding a delay to the end of the line. However, this results in the script crashing/freezing after a couple of trials, or if the animal remains in the conditional True part of the corridor. The SubscribeSubject (z pos) is updated very frequently, and my feeling is that the delay is not causing the process to be skipped during the delay period, but rather queued up in some way until the script eventually freezes.

Do you have any advice on how such a function could be implemented?

Thanks! 
<img width="581" alt="Screenshot 2024-11-19 at 1 42 01 PM" src="https://github.com/user-attachments/assets/f8389eb7-c2e4-4c97-9691-bcbec37c981a">

**Comment by bruno-f-cruz** [comment]
Add your workflow please (#864)

If `z pos` is a PublishSubjet you can simply use repeat, otherwise you can also use GateInterval:

![image](https://github.com/user-attachments/assets/f4d351a7-cb97-4e9e-847e-a9ac72c6c73a)
